mixins/interactive/interactive-mixin.js

```diff
@@ -106,10 +107,15 @@ export const InteractiveMixin = superclass => class extends LocalizeCoreElement(
 	}
 
 	async _handleInteractiveKeyDown(e) {
-		if (this._interactive && e.keyCode === keyCodes.ESCAPE) {
+		if (!this._interactive) return;
+
+		if (e.keyCode === keyCodes.ESCAPE && !e.target?.opened) {
```

Comment:
The [DropdownContentMixin](https://github.com/BrightspaceUI/core/blob/main/components/dropdown/dropdown-content-mixin.js#L663) relies on [dismissible.js](https://github.com/BrightspaceUI/core/blob/main/helpers/dismissible.js) to help manage the dismiss behaviour. It works best when everything uses it because it manages a simple stack of callbacks - invoking the top callback when the user presses the `[Esc]` key. 

The `InteractiveMixin` currently doesn't use `dismissible.js`, but perhaps it could. It would be set up the same as dropdowns... `setDismissible` would be called when the use enters interactive mode (like we do when a dropdown is opened), and `clearDismissible` would be called when the user leaves interactive mode. I think that would avoid the need for `opened` stuff creeping in here.

Thoughts? I think `dismissible.js` existed before `InteractiveMixin`, and so I'd be surprised if this wasn't considered at the time. There might be a reason using it wasn't possible.

Reply:
Good idea! So far this looks to work well